ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockHash.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.MinerDataResult;
+import org.hyperledger.besu.ethereum.api.query.BlockWithMetadata;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.api.query.TransactionReceiptWithMetadata;
+import org.hyperledger.besu.ethereum.api.query.TransactionWithMetadata;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import org.apache.tuweni.units.bigints.BaseUInt256Value;
+
+public class EthGetMinerDataByBlockHash implements JsonRpcMethod {
+  private final Supplier<BlockchainQueries> blockchain;
+  private final ProtocolSchedule protocolSchedule;
+
+  public EthGetMinerDataByBlockHash(
+      final BlockchainQueries blockchain, final ProtocolSchedule protocolSchedule) {
+    this(Suppliers.ofInstance(blockchain), protocolSchedule);
+  }
+
+  public EthGetMinerDataByBlockHash(
+      final Supplier<BlockchainQueries> blockchain, final ProtocolSchedule protocolSchedule) {
+    this.blockchain = blockchain;
+    this.protocolSchedule = protocolSchedule;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_GET_MINER_DATA_BY_BLOCK_HASH.getMethodName();
+  }
+
+  /*
+  Getting static block reward
+  pass in the block height (as a long) get the block reward (as wei)
+  each block spec has it.  but it’s only meaningful w/in 6 blocks of the fork.
+
+  Didn’t happen at the two places we reduced block reward on mainnet.  I say leave a comment but write no code.
```

Comment:
Even though this is a slack quote it should be in a documentary tone.  I'd also move it around 108-111 where the uncle reward calculation is calculated as it is a code relevant comment not a quasi-javadoc comment.

Reply:
Comment removed as this is no longer relevant.